ERROR in The Angular Compiler requires TypeScript >=3.6.4 and <3.7.0 but 3.4.3 was found instead.

## Angular CLI version

```txt
Angular CLI: 9.0.0-rc.3
Node: 10.16.3
OS: win32 x64
Angular: 9.0.0-rc.3
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-rc.3
@angular-devkit/build-angular     0.900.0-rc.3
@angular-devkit/build-optimizer   0.900.0-rc.3
@angular-devkit/build-webpack     0.900.0-rc.3
@angular-devkit/core              9.0.0-rc.3
@angular-devkit/schematics        9.0.0-rc.3
@ngtools/webpack                  9.0.0-rc.3
@schematics/angular               9.0.0-rc.3
@schematics/update                0.900.0-rc.3
rxjs                              6.5.3
typescript                        3.4.3
webpack                           4.41.2
```

## Windows version

```
Microsoft Windows [ver 10.0.18363.476]
```

## Reproduce steps

```sh
git clone https://github.com/mgechev/angular-realworld-example-app-qucklink
cd angular-realworld-example-app-qucklink
npm install
ng build --prod
```

## Actual error

```
ERROR in The Angular Compiler requires TypeScript >=3.6.4 and <3.7.0 but 3.4.3 was found instead.
```

![image](https://user-images.githubusercontent.com/88981/69301926-d0e8f100-0c52-11ea-8203-7bc372fe4855.png)

After upgrade typescript to `3.6`, I got the new error.

```sh
npm i typescript@3.6
ng build --prod
```

```txt
ERROR in ../node_modules/ngx-quicklink/link.directive.d.ts:16:30 - error TS2307: Cannot find module '@angular/router/src/url_tree'.       
16     readonly urlTree: import("@angular/router/src/url_tree").UrlTree;
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

![image](https://user-images.githubusercontent.com/88981/69302020-2a512000-0c53-11ea-9da3-5bb63806d103.png)

--- a/e2e/tsconfig.e2e.json
+++ b/e2e/tsconfig.e2e.json
@@ -3,8 +3,8 @@
   "compilerOptions": {
     "outDir": "../out-tsc/e2e",
     "baseUrl": "./",
-    "module": "commonjs",
-    "target": "es5",
+    "module": "esnext",
+    "target": "es2015",
     "types": [
       "jasmine",
       "jasminewd2",

--- a/package.json
+++ b/package.json
@@ -27,7 +27,7 @@
     "core-js": "3",
     "marked": "^0.6.2",
     "moment": "^2.24.0",
-    "ngx-quicklink": "^0.0.11",
+    "ngx-quicklink": "^0.1.5",
     "rxjs": "^6.4.0",
     "rxjs-compat": "^6.4.0",
     "zone.js": "~0.9.0"

--- a/package.json
+++ b/package.json
@@ -52,6 +52,6 @@
     "protractor": "~5.4.2",
     "ts-node": "~8.1.0",
     "tslint": "~5.16.0",
-    "typescript": "3.4.3"
+    "typescript": "3.6.4"
   }
 }

--- a/src/app/app-routing.module.ts
+++ b/src/app/app-routing.module.ts
@@ -5,19 +5,19 @@ import { QuicklinkStrategy } from 'ngx-quicklink';
 const routes: Routes = [
   {
     path: 'settings',
-    loadChildren: './settings/settings.module#SettingsModule'
+    loadChildren: () => import('./settings/settings.module').then(m => m.SettingsModule)
   },
   {
     path: 'profile',
-    loadChildren: './profile/profile.module#ProfileModule'
+    loadChildren: () => import('./profile/profile.module').then(m => m.ProfileModule)
   },
   {
     path: 'editor',
-    loadChildren: './editor/editor.module#EditorModule'
+    loadChildren: () => import('./editor/editor.module').then(m => m.EditorModule)
   },
   {
     path: 'article',
-    loadChildren: './article/article.module#ArticleModule'
+    loadChildren: () => import('./article/article.module').then(m => m.ArticleModule)
   }
 ];
 

--- a/src/tsconfig.app.json
+++ b/src/tsconfig.app.json
@@ -1,10 +1,10 @@
 {
   "extends": "../tsconfig.json",
   "compilerOptions": {
-    "target": "es2015",
+    "target": "esnext",
     "outDir": "../out-tsc/app",
     "baseUrl": "./",
-    "module": "es2015",
+    "module": "esnext",
     "types": []
   },
   "exclude": [

--- a/src/tsconfig.spec.json
+++ b/src/tsconfig.spec.json
@@ -3,8 +3,8 @@
   "compilerOptions": {
     "outDir": "../out-tsc/spec",
     "baseUrl": "./",
-    "module": "commonjs",
-    "target": "es5",
+    "module": "esnext",
+    "target": "es2015",
     "types": [
       "jasmine",
       "node"

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -7,7 +7,7 @@
     "moduleResolution": "node",
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
-    "target": "es5",
+    "target": "es2015",
     "typeRoots": [
       "node_modules/@types"
     ],

